src/stacked_borrows.rs

```diff
@@ -269,16 +268,24 @@ impl<'tcx> Stack {
     fn check_protector(item: &Item, tag: Option<Tag>, global: &GlobalState) -> InterpResult<'tcx> {
         if let Tag::Tagged(id) = item.tag {
             if Some(id) == global.tracked_pointer_tag {
-                throw_machine_stop!(TerminationInfo::PoppedTrackedPointerTag(item.clone()));
+                register_err(
+                    InterpError::MachineStop(Box::new(TerminationInfo::PoppedTrackedPointerTag(
+                        item.clone(),
+                    )))
+                    .into(),
+                );
             }
         }
         if let Some(call) = item.protector {
             if global.is_active(call) {
                 if let Some(tag) = tag {
-                    throw_ub!(UbExperimental(format!(
-                        "not granting access to tag {:?} because incompatible item is protected: {:?}",
-                        tag, item
-                    )));
+                    return Err(err_ub_experimental(
```

Comment:
Why this diff? Why should this even build, `err_ub_experimental` just takes a string...?

Reply:
There's more of this weird diff further down.

Reply:
Oh, you added a new function. But you added it at the end, after its uses, so there's no way one could know that when doing a review. See why I ask you to put definitions before uses?^^

I am not sure I like this... it makes the error message really long, and these are already too long. Also the function is badly named as it is not for general `UbExperimental` errors.

Reply:
> it makes the error message really long

I can make it add its own `note:` entry

Reply:
Not easily though, that would have to happen when the error is emitted, not here where the error is created, right?

Reply:
yes, but since we're explicitly doing this in `process_errors`, we can forward a flag to the reporting function telling it how to emit. We can even encode the error/warning level in the static storage

Reply:
You can't actually use `UbExperimental` for that though, I think? That just doesn't have enough flexibility to pass information to the error reporting function.

Reply:
True. I'd need to use `MachineStop`. I'll remove the flag for now and let users discover the feature via the documentation